api/api.go

```diff
@@ -249,6 +255,28 @@ func (h *caHandler) Route(r Router) {
 	r.MethodFunc("GET", "/federation", h.Federation)
 	// For compatibility with old code:
 	r.MethodFunc("POST", "/re-sign", h.Renew)
+
+	ah := h.acmeHandler
+	// Standard ACME API
+	r.MethodFunc("GET", ah.Dir.NewNonce, ah.addNonce(ah.GetNonce))
+	r.MethodFunc("GET", ah.Dir.Directory, ah.addNonce(ah.GetDirectory))
+
+	extractPayloadByJWK := func(next nextHTTP) nextHTTP {
+		return ah.addNonce(ah.addDirectory(ah.verifyContentType(ah.parseJWS(ah.validateJWS(ah.extractJWK(ah.verifyAndExtractJWSPayload(next)))))))
+	}
+	extractPayloadByKid := func(next nextHTTP) nextHTTP {
+		return ah.addNonce(ah.addDirectory(ah.verifyContentType(ah.parseJWS(ah.validateJWS(ah.lookupJWK(ah.verifyAndExtractJWSPayload(next)))))))
+	}
+
+	r.MethodFunc("POST", ah.Dir.NewAccount, extractPayloadByJWK(ah.NewAccount))
+	r.MethodFunc("POST", ah.Dir.GetAccount("{accID}", false), extractPayloadByKid(ah.UpdateAccount))
+	r.MethodFunc("POST", ah.Dir.NewOrder, extractPayloadByKid(ah.NewOrder))
+	r.MethodFunc("POST", ah.Dir.GetOrder("{ordID}", false), extractPayloadByKid(ah.GetOrder))
+	r.MethodFunc("POST", ah.Dir.GetFinalize("{ordID}", false), extractPayloadByKid(ah.FinalizeOrder))
+	//r.MethodFunc("POST", ah.Dir.GetFinalize("{ordID}"), extractPayloadByKid(ah.Finalize))
+	r.MethodFunc("POST", ah.Dir.GetAuthz("{authzID}", false), extractPayloadByKid(ah.GetAuthz))
+	r.MethodFunc("POST", ah.Dir.GetChallenge("{chID}", false), extractPayloadByKid(ah.GetChallenge))
+	r.MethodFunc("POST", ah.Dir.GetCertificate("{certID}", false), extractPayloadByKid(ah.GetCertificate))
```

Comment:
To split the code better we can Add to the RouterHandler interface the method `Route(pattern string, fn func(r Router)) Router` and use have a new Route method in `api/acme.go`, more or less in the same way that we add the `/1.0` prefix in https://github.com/smallstep/certificates/blob/master/ca/ca.go#L106-L108

So the routes will become for example `/acme/directory` or `/1.0/acme/directory`